fw/http_sess.c

```diff
@@ -588,17 +588,19 @@ tfw_http_sess_resp_process(TfwHttpResp *resp, bool cache)
 	{
 		return 0;
 	}
-	BUG_ON(!req->sess);
-
-	/*
-	 * RFC 6265 4.1.1 and 4.1.2 says that we should not set session cookie
-	 * if it's not necessary. Since client didn't send up the cookie and
-	 * it seems that we don't enforce them, we can just set the cookie in
-	 * each response forwarded to the client.
-	 */
-	if (test_bit(TFW_HTTP_B_HAS_STICKY, req->flags))
-		return 0;
-	return tfw_http_sticky_add(resp, cache);
+	if (req->need_jsch) {
```

Comment:
This function is called for all types of cookies which are set by tempesta. It is not only about js_challenge, so this check is wrong